Update from source

hcarver opened the pull request.

dylandrop committed:
init accounts

jrmyward committed:
Adds support for customer coupons.

* This is mostly borrowed from #74
* Fixes #72

jrmyward committed:
Adds support for customer coupons.

* This is mostly borrowed from #74
* Fixes #72

nhocki committed:
Fix customer_examples for new Stripe API.

nhocki committed:
Fix charge token examples and customer card examples.

nhocki committed:
Fix subscription handler specs.

nhocki committed:
Fixing cards examples.

Had to add the old customer cards endpoints to support them while Stripe
decides what's gonna happen there... See stripe/stripe-ruby#215

nhocki committed:
Fix charges examples.

nhocki committed:
Fix invoice examples

nhocki committed:
Use the latest version of Stripe

This version has the endpoints problems fixed.

gilbert committed:
Merge branch 'stripe_api_update' of https://github.com/nhocki/stripe-ruby-mock into nhocki-stripe_api_update

gilbert committed:
Merge branch 'nhocki-stripe_api_update'

gilbert committed:
Bump version to 2.1.0

gilbert committed:
Add note about versioning

theoretick committed:
Minor charge param requirements refactor

theoretick committed:
Require amount and currency for Stripe::Charge.create

theoretick committed:
Merge branch 'master' of github.com:rebelidealist/stripe-ruby-mock into add-charge-positive-integer-error-failure

Conflicts:
	spec/shared_stripe_examples/card_token_examples.rb

TrevorHinesley committed:
Fixed add_card_to_object method to actually replace card when needed

TrevorHinesley committed:
Updated spec

gilbert committed:
Merge pull request #195 from TrevorHinesley/master

Fixed add_card_to_object method to actually replace card when needed

gilbert committed:
Create Plan does not take statement_description

gilbert committed:
Update invalid request error to reflect Stripe's

gilbert committed:
Merge branch 'add-charge-positive-integer-error-failure' of https://github.com/theoretick/stripe-ruby-mock into theoretick-add-charge-positive-integer-error-failure

gilbert committed:
Revert "Update invalid request error to reflect Stripe's"

This reverts commit f4414ae24ff181afa03d64c931508d20bbf736eb.

gilbert committed:
Properly update invalid request errors to reflect Stripe's

gilbert committed:
Add live tests to TravisCI builds

gilbert committed:
Add live tests to TravisCI builds

gilbert committed:
Lock in stripe-ruby version to 1.20.1 until we fix 1.20.2

gilbert committed:
Bump version to 2.1.1

jasdeepgosal committed:
Updated default webhook fixtures to use “source” instead of “card”

Based on the 2015-02-18 API changes: https://stripe.com/docs/upgrades#2015-02-18

nathanpalmer committed:
Added support for the card parameter of a token to be a string
Added a test to emulate the way we use it (which is creating a customer on two different accounts using stripe connect)

jwilsjustin committed:
Update README to reflect :source param

ccahoon committed:
Implement saving already existing Charges

Clean up

alexmamonchik committed:
tests of coupons. Live tests passed

alexmamonchik committed:
tests of coupons. Live tests passed

Commit:
added accounts test and merge with PR#218

Commit:
merge with wanelo's accounts

Commit:
added accounts test and merge with PR#218

alexmamonchik committed:
Coupons support for subscriptons

gilbert committed:
Update 2.1.x in travis file

gilbert committed:
Use separate keys for each ruby build

gilbert committed:
Merge pull request #201 from rebelidealist/live-builds

Add live tests to TravisCI builds

gilbert committed:
Merge pull request #211 from jwilsjustin/patch-1

Update README to reflect :source param

gilbert committed:
Add test keys to travis.yml (what could possibly go wrong?)

gilbert committed:
Merge pull request #220 from ccahoon/update_charges

Implement saving already existing Charges

gilbert committed:
Add request for more contributors

alexmamonchik committed:
ignore idea files

alexmamonchik committed:
fixed tests for validation of plans

gilbert committed:
Merge pull request #222 from alexmamonchik/plan-validation-error

fixed tests for validation of plans

alexmamonchik committed:
Fix error with assert_existance

alexmamonchik committed:
Remove created coupons after live testing

alexmamonchik committed:
Merge pull request #223 from alexmamonchik/customer-coupon

customer coupons support

alexmamonchik committed:
Clear coupons for live testing

alexmamonchik committed:
Merge pull request #214 from alexmamonchik/discount-for-subscription

Fixes #197 and #79

alexmamonchik committed:
refactoring coupons live tests

alexmamonchik committed:
Merge pull request #225 from rebelidealist/live-strategy-for-coupons

refactoring live tests for coupons

alexmamonchik committed:
Fix #226

alexmamonchik committed:
Merge pull request #228 from rebelidealist/card-source

Fix #226 Creating Customer with Card hash as source

Commit:
set api_key on all requests

https://github.com/rebelidealist/stripe-ruby-mock/issues/209

alexmamonchik committed:
Fix #224

alexmamonchik committed:
Merge pull request #229 from rebelidealist/transfer-amount

Fix #224

alexmamonchik committed:
set stripe to 1.21.0 in gemspec

alexmamonchik committed:
Fix api_key must be a string error

alexmamonchik committed:
fix tests for card_token

alexmamonchik committed:
Fix tests for customers

alexmamonchik committed:
fix invoice tests

alexmamonchik committed:
Fix #230

alexmamonchik committed:
Merge pull request #231 from rebelidealist/list-of-charges

Fix #230

alexmamonchik committed:
Include commits from  #218 and #55

Commit:
Added has_more to Stripe::ListObject JSON for Transfer wehooks

alexmamonchik committed:
Merge pull request #233 from neilbartley/transfer_has_more

has_more on Stripe::ListObject

j15e committed:
Clarify warning is from StripeMock

j15e committed:
Show params only if present

gilbert committed:
Merge pull request #236 from didacte/feature/improved-warnings

Feature/improved warnings

kathyonu committed:
3xexample for live Stripe tests

alexmamonchik committed:
Merge pull request #239 from kathyonu/rspec3xexample

Fix for #235 Documentation for live Stripe tests

saidM committed:
renamed card to source

alexmamonchik committed:
Merge pull request #240 from saidM/card_to_source

Renamed card to source inside webhooks

alexmamonchik committed:
cleanup

alexmamonchik committed:
retrieve all accounts

tzzzoz committed:
rename card to source in README.md

gilbert committed:
Merge pull request #241 from rebelidealist/accounts

Accounts

gilbert committed:
Update test to reflect live server

gilbert committed:
Merge pull request #247 from rebelidealist/error-msg

Update test to reflect live server

fakeleft committed:
Fixed config var name.

alexmamonchik committed:
Merge pull request #248 from fakeleft/master

Fixed config var name.

george-carlin committed:
Change smelly "if == false" to "unless

alexmamonchik committed:
Merge pull request #249 from georgemillo/patch-1

Boyscouting

antoinelyset committed:
Allow Customer creation without card but with trial_end

alexmamonchik committed:
Merge pull request #251 from antoinelyset/customer-trial-end

Allow Customer creation without card but with trial_end

antoinelyset committed:
Allow Subscription update without card but with trial_end

alexmamonchik committed:
Merge pull request #255 from antoinelyset/subscription-trial-end

Allow Subscription update without card but with trial_end

alexmamonchik committed:
remove fee fields from charge mock

alexmamonchik committed:
fix subscriptions tests

louim committed:
Fix test errors on charges and oauth

- Workaround for unwanted params that Stripe sends even when they aren't
modified un Update Charge call.
- Fix oauth tests being run even when the oauth => true isn't set.
- Use let instead of instance variable in oauth tests.
- Allow tests to be focused to debug.

alexmamonchik committed:
update to stripe 1.27.0

alexmamonchik committed:
Merge pull request #245 from tzzzoz/master

rename card to source in README.md

alexmamonchik committed:
return json_body for CardError

alexmamonchik committed:
Merge pull request #261 from rebelidealist/bug/card-error

return json_body for CardError

gilbert committed:
Merge pull request #262 from rebelidealist/newest

Update Stripe gem to latest version

gilbert committed:
Bump version to 2.2.0

louim committed:
Target most recent version of the Stripe API.

louim committed:
Delete empty params before merge to prevent overwrite

Without deleting, the params would overwrite the `sources` list, wich
would lose it's `object: 'list'` key and would be converted to a normal
Stripe::Object instead of a Stripe::ListObject

alexmamonchik committed:
Merge pull request #266 from didacte/bugfix/customer-assign-default-card

Delete empty params before merge to prevent overwrite

Commit:
Testing fix for customer source update

Commit:
Fix typo

Commit:
DRY up changes

Commit:
Add spec for new customer source update behavior

Commit:
Fix format error with source expiration year and month

Commit:
Make generation of fingerprint dependent on live mode status

alexmamonchik committed:
Locked stripe version to 1.28.1 while opened #270

alexmamonchik committed:
ignore rbenv file

alexmamonchik committed:
Merge pull request #269 from SwedenUnltd/update-customer-source

Update customer source from hash

alexmamonchik committed:
Merge pull request #264 from louim/patch-1

Target most recent version of the Stripe API.

alexmamonchik committed:
Merge pull request #208 from nathanpalmer/token-card-string-support

Added support for the card parameter of a token to be a string

alexmamonchik committed:
added oAuth test when card parameter of a token to is a string

alexmamonchik committed:
fixed no implicit conversion of Symbol into Integer

alexmamonchik committed:
Merge pull request #272 from rebelidealist/fix/transfers

fixed no implicit conversion of Symbol into Integer #270

alexmamonchik committed:
update stripe to 1.31.0

alexmamonchik committed:
Merge branch 'master' of github.com:rebelidealist/stripe-ruby-mock

gilbert committed:
Bump version to 2.2.1

Commit:
Add support for Orders

ball-hayden committed:
Prevent metadata being lost on charge update

alexmamonchik committed:
Merge pull request #279 from slash7/add_orders

Add support for Orders

ball-hayden committed:
Correct spelling

alexmamonchik committed:
Merge pull request #280 from SparkSeat/fix-charge-update-loss

Prevent metadata being lost on charge update

benjie committed:
Add failing test for saving a customer not breaking sources.

benjie committed:
Fix empty sources.

benjie committed:
Handle dud subscriptions too.

benjie committed:
Restore total_count functionality.

sublimecoder committed:
create a charge object when adding a subscription to a customer

nomanurrehman committed:
Add disputes class with empty methods

nomanurrehman committed:
Add request handlers for disputes

nomanurrehman committed:
Add disputes module to server instance

nomanurrehman committed:
Add disputes examples

nomanurrehman committed:
Add disputes mock data

nomanurrehman committed:
Add disputes module

nomanurrehman committed:
Add disputes examples

nomanurrehman committed:
Add close dispute and update dispute request handlers

nomanurrehman committed:
Add dispute mock data

nomanurrehman committed:
Add method to mock disputes on server startup

nomanurrehman committed:
Add mocking disputes on server startup

nomanurrehman committed:
Clean up dispute endpoints implementation

nomanurrehman committed:
Add test cases for disputes

Close dispute test case is failing

nomanurrehman committed:
Fix dispute end point handlers sequence

nomanurrehman committed:
Add test cases for invalid dispute

TrevorHinesley committed:
Added application_fee to invoices

TrevorHinesley committed:
Merge remote-tracking branch 'upstream/master'

TrevorHinesley committed:
Added description to Data.mock_invoice

TrevorHinesley committed:
Added metadata to Data.mock_invoice

TrevorHinesley committed:
Added receipt_number to Data.mock_invoice

TrevorHinesley committed:
Added statement_descriptor to Data.mock_invoice

TrevorHinesley committed:
Added tax to Data.mock_invoice

TrevorHinesley committed:
Added tax_percent to Data.mock_invoice

TrevorHinesley committed:
Added webhooks_delivered_at to Data.mock_invoice

alexmamonchik committed:
Merge pull request #285 from sublimecoder/master

create a charge object when adding a subscription to a customer

alexmamonchik committed:
use newest version of bundler

alexmamonchik committed:
Merge branch 'master' of github.com:rebelidealist/stripe-ruby-mock

TrevorHinesley committed:
Merge remote-tracking branch 'upstream/master'

nomanurrehman committed:
Merge branch 'master' of https://github.com/rebelidealist/stripe-ruby-mock

alexmamonchik committed:
Merge pull request #289 from TrevorHinesley/master

Added missing Invoice fields

alexmamonchik committed:
Merge pull request #287 from nomanurrehman/master

Dispute Endpoints Implemented

nomanurrehman committed:
Add api endpoint to retrieve all events

nomanurrehman committed:
Add test cases for all events api endpoint

TrevorHinesley committed:
Added discountable and subscription to Data.mock_line_item

TrevorHinesley committed:
Added discountable to get_mock_subscription_line_item

TrevorHinesley committed:
Merge remote-tracking branch 'upstream/master'

Commit:
Update card_helpers.rb

cards_or_sources[:total_count] || 0 + 1 -> 

if cards_or_sources[:total_count] == 0
  return 0
else 
  return 0 + 1 # return 1
end

I believe the meaning was:

 if cards_or_sources[:total_count] == 0 || cards_or_sources[:total_count].nil?
 return 1
else
return cards_or_sources[:total_count]  + 1
end

TrevorHinesley committed:
Added application_fee to Data.mock_charge

TrevorHinesley committed:
Added destination to Data.mock_charge

TrevorHinesley committed:
Added fraud_details to Data.mock_charge

TrevorHinesley committed:
Added receipt_email and receipt_number to Data.mock_charge

TrevorHinesley committed:
Added shipping to Data.mock_charge

TrevorHinesley committed:
Added statement_descriptor to Data.mock_charge

TrevorHinesley committed:
Added transfer to Data.mock_charge

alexmamonchik committed:
Merge pull request #291 from nomanurrehman/master

All Events Endpoint Implemented

alexmamonchik committed:
Merge pull request #292 from TrevorHinesley/master

Added missing fields to mocked invoices

alexmamonchik committed:
Merge pull request #293 from liorsion/patch-1

Update card_helpers.rb

alexmamonchik committed:
Merge pull request #283 from benjie/fix-customer-with-sources-save

Fix for saving a customer invalidating sources

pcboy committed:
Hotfix: return correct error code after kill_server

alexmamonchik committed:
Merge pull request #297 from pcboy/hotfix/correct-return-code-at-server-exit

Hotfix: return correct error code after kill_server

alexmamonchik committed:
fixed error message when invalid card number

jasdeepgosal committed:
Added support for / fixed a bug with `charge.mark_as_safe`

Previously, calling `charge.mark_as_safe` would blow up [here](https://github.com/charitywater/stripe-ruby-mock/blob/eaf7f5f30b92f32dc40ef47d67cff2ac093536ae/lib/stripe_mock/request_handlers/charges.rb#L42) because `allowed_params` was returning `nil`.

Commit:
Add `currency` attribute to customer

Commit:
Raise error if trying to combine currencies on the same customer.

alexmamonchik committed:
Merge pull request #207 from charitywater/card-to-source

Updated default webhook fixtures to use “source” instead of “card”

alexmamonchik committed:
fixed #305

Commit:
Add currency specs

mhenrixon committed:
Allow webhooks for Stripe Connect

Stripe uses the user_id parameter when the webhook if for Connect. This
commit extracts user_id from the passed in params and sets it at the
root level of the json. This should be safe enough as there is no other
use of user_id in the stripe api.

alexmamonchik committed:
Merge pull request #308 from mhenrixon/master

Allow webhooks for Stripe Connect

alexmamonchik committed:
Merge pull request #307 from slash7/add_customer_currency

Added `currency` attribute to customer

alexmamonchik committed:
Merge pull request #304 from charitywater/mark-as-safe-bug

Added support for / fixed a bug with `charge.mark_as_safe`

gilbert committed:
Bump version to 2.2.2

alexmamonchik committed:
fix for #278

alexmamonchik committed:
use stripe 1.31.0 or more

ehsanul committed:
allow for updating an application fee upon charge capture

alexmamonchik committed:
fix for #319

shawnonthenet committed:
Update stripe-ruby-mock.gemspec

Missing space was causing the match to be exactly version 1.31.0 of stripe gem.  Added space to make it match >= 1.31.0

alexmamonchik committed:
Merge pull request #319 from MelianLabs/capture_with_application_fee

allow for updating an application fee upon charge capture

alexmamonchik committed:
Merge pull request #320 from shawnonthenet/master

Update stripe-ruby-mock.gemspec

Commit:
Add Dummy key

alexmamonchik committed:
Merge pull request #322 from lucas-aragno/master

Avoid to require an api key to get the instance working

Rabsztok committed:
Make plan_id when creating subscription string/symbol agnostic

alexmamonchik committed:
Merge pull request #323 from Rabsztok/master

Make plan_id when creating subscription string/symbol agnostic

antoniobg commented:
Hey @hcarver, there are some conflicts and they're related with the changes you made in this commits:

https://github.com/antoniobg/stripe-ruby-mock/commit/0c6ec855e1d85e9baeb28eaa6e3e8208e741336c
https://github.com/antoniobg/stripe-ruby-mock/commit/eb062e8e053fe46a3359ae7dfa583d5ef574cac1

hcarver commented:
Ah, looking at their commits, I think it's fine to just overwrite the changes I made. In fact, I'm going to try just using the original gem to see if it "just works" now.

lfittl committed:
Implement basic support for bank account sources.

This allows creating a customer using bank account tokens, defaulting
to "new" state, and then verifying when verify endpoint gets called.

alexmamonchik committed:
Merge pull request #325 from lfittl/basic-bank-account-support

Implement basic support for bank account sources

refacrot committed:
Get card from customer if card not specified on new charge

Willianvdv committed:
Using Drb for remote error queue

Willianvdv committed:
Share the error queue

gilbert committed:
Bump version to 2.2.3

gilbert committed:
Bump version to 2.2.3

typeoneerror committed:
Locks stripe to < 1.42

1.42 introduces a new subscriptions API, so the subscriptions request
handler will need to be updated before we can include that version.

alexmamonchik committed:
Merge pull request #329 from typeoneerror/master-blaster

Master Blaster: Fix commits lost by force push, lock Stripe to < 1.42

typeoneerror committed:
Adds basics for BalanceTransaction mocking

gilbert committed:
Bump version to 2.2.4

alexmamonchik committed:
Merge pull request #330 from typeoneerror/balance-transaction-mocking

Adds basics for BalanceTransaction mocking

Willianvdv committed:
Remove deps

Willianvdv committed:
Undo changes

Commit:
Add status and receipt_number to refund response

alexmamonchik committed:
Merge pull request #337 from dorkrawk/master

Add status and receipt_number to refund response

Empact committed:
Enable generating card and bank tokens without arguments

Via StripeMock.generate_card_token and StripeMock.generate_bank_token.

Because sometimes you don't care about the token, so long as it is one.

Empact committed:
Support generate_bank_token in stripe_helper

This mostly adapts existing card flows to more general sources flows, and
adds bank account support along the way.

This is not a complete solution, but it improves support significantly.

alexmamonchik committed:
Merge pull request #338 from Nootrobox/generate-bank-token

Support generate_bank_token in stripe_helper

alexmamonchik committed:
Merge pull request #339 from Nootrobox/default-token-values

Enable generating card and bank tokens without arguments

alexmamonchik committed:
Merge pull request #328 from JakovlevDS/fix/charge-source-from-customer

Get card from customer if card not specified on new charge

alexmamonchik committed:
Merge pull request #327 from Willianvdv/srb

Adds remote error queue and replaces Jimson with Ruby's DRb

pwim committed:
Remove unnecessary dependency on jimson

With f3be892bf69c24e9da4f8f4012e8c691ed2716db, it is no longer needed.

multi_json was implicitly being depended on by this gem via jimson, so
make the dependency explicit instead.

adbjesus committed:
Update gemspec

adbjesus committed:
Update to 1.42 done

adbjesus committed:
Update to 1.43

adbjesus committed:
Updated to include cvc since is stated as usually required in the API

alexmamonchik committed:
Merge pull request #342 from pwim/drop-jimson

Remove unnecessary dependency on jimson

adbjesus committed:
(re)add still existing customer/subscriptions routes

vadzimpiatkevich committed:
Fix typo in generating bank token example

vadzimpiatkevich committed:
Use "case" statement for conditional branching

alexmamonchik committed:
Merge pull request #345 from vadzimpiatkevich/bank-account-token-mocking-spec

Fix typo in generating bank token example

alexmamonchik committed:
Merge pull request #346 from vadzimpiatkevich/bank-and-card-token-mock-refactoring

Use "case" statement for conditional branching

alexmamonchik committed:
Merge pull request #343 from adbjesus/update-to-1_42

Update to 1.43

vadzimpiatkevich committed:
Add metadata to plan object

add metadata field;
change the order of the fields in accordance with the stripe api;
fix subscription request handler according to the chnges.

gilbert committed:
Bump version to 2.3.0

vadzimpiatkevich committed:
Compress if/else statement to the ternary operator

alexmamonchik committed:
Merge pull request #347 from vadzimpiatkevich/plan-metadata

Add metadata to plan object

vadzimpiatkevich committed:
Fix assigning coupon to a subscription

Assigning coupon to a subscription actually updates coupon (there was incorrect behavior when it always returns the default mock coupon).

vadzimpiatkevich committed:
Fix assigning coupon when subscription creating

vadzimpiatkevich committed:
Resolve gem dependencies conflict

Build failed because Travis CI can't install rack (2.0.1) that is dependency of thin.
It happens because rack (2.0.1) requires Ruby version >= 2.2.2. But Travis works with higher versions.
I downgrade thin version to 1.6.4 that just requires ruby >= 1.8.5.

alexmamonchik committed:
Merge pull request #352 from vadzimpiatkevich/assigning-coupon-to-a-subscription

Fix assigning coupon to a subscription

vadzimpiatkevich committed:
Fix TypeError on customer subscriptions retrieve

alexmamonchik committed:
Merge pull request #353 from vadzimpiatkevich/customer-subscriptions-retrieve

Fix TypeError on customer subscriptions retrieve

alexmamonchik committed:
fix for #354

alexmamonchik committed:
Merge pull request #361 from rebelidealist/fix/subscription-delete

fix for #354
